src/routes/safe/components/Apps/components/AppsList.tsx

```diff
@@ -110,40 +158,87 @@ const AppsList = (): React.ReactElement => {
-                <StyledLink key={a.url} to={`${appsPath}?appUrl=${encodeURI(a.url)}`}>
-                  <AppCard isLoading={isAppLoading(a)} iconUrl={a.iconUrl} name={a.name} description={a.description} />
-                </StyledLink>
-                {isCustomApp(a.url, appList) && (
-                  <IconBtn
-                    title="Remove"
-                    onClick={(e) => {
-                      e.stopPropagation()
-
-                      setAppToRemove(a)
-                    }}
-                  >
-                    <Icon size="sm" type="delete" color="error" />
-                  </IconBtn>
-                )}
-              </AppContainer>
-            ))}
-        </CardsWrapper>
-
-        <IconText
-          color="secondary"
-          iconSize="sm"
-          iconType="info"
-          text="These are third-party apps, which means they are not owned, controlled, maintained or audited by Gnosis. Interacting with the apps is at your own risk. Any communication within the Apps is for informational purposes only and must not be construed as investment advice to engage in any transaction."
-          textSize="sm"
-        />
+        <SearchCard>
+          <MuiTextField
+            InputProps={{
+              'aria-label': 'search',
+              startAdornment: <SearchIcon />,
+              endAdornment: appSearch && (
+                <IconButton onClick={() => setAppSearch('')}>
+                  <ClearIcon />
+                </IconButton>
+              ),
+            }}
+            onChange={(event) => setAppSearch(event.target.value)}
+            placeholder="Search…"
+            value={appSearch}
+            style={{ width: '100%' }}
+          />
+        </SearchCard>
+        {noAppsFound && (
+          <>
+            <NoAppsFoundTextContainer>
+              <InfoIcon />
+              <Text size="xl">
+                No apps found matching <b>{appSearch}</b>
+              </Text>
+            </NoAppsFoundTextContainer>
+            <ClearSearchButton size="md" color="primary" variant="contained" onClick={() => setAppSearch('')}>
+              <Icon type="cross" size="sm" />
+              <Text size="xl" color="white">
+                Clear search
+              </Text>
+            </ClearSearchButton>
+          </>
+        )}
+
+        <AnimatePresence>
+          <CardsWrapper>
+            {!appSearch && (
+              <AppCard iconUrl={AddAppIcon} onClick={openAddAppModal} buttonText="Add custom app" iconSize="lg" />
+            )}
+            {apps
+              .filter((a) => a.fetchStatus !== SAFE_APP_FETCH_STATUS.ERROR)
+              .map((a) => (
+                <AppContainer
+                  key={a.url}
+                  layout
+                  initial={{ opacity: 0 }}
+                  animate={{ opacity: 1 }}
+                  exit={{ opacity: 0 }}
+                >
+                  <StyledLink key={a.url} to={`${appsPath}?appUrl=${encodeURI(a.url)}`}>
```

Comment:
Unrelated but could you please change the key to `a.id` (for custom apps id could remain the URL).